### PR TITLE
EKF: model bluff body drag vs angle of sideslip using an elliptical distribution

### PR DESCRIPTION
**Pull request by priseborough**
Refer to https://discuss.px4.io/t/potential-bug-in-parasite-drag-formula/28544/6 for a more detailed discussion about the modelling of bluff body drag that this pull request modifies.

The legacy code models bluff body drag using a method that results in a variation in drag with angle of sideslip that  resembles a diamond shape, not an ellipse, eg.

Screen Shot 2022-08-16 at 6 15 53 pm

This change does the bluff body drag calculation in polar coordinates using an elliptical drag coefficient distribution. The observation Jacobian is current calculated using the total airspeed which is likely to make wind states change more rapidly for combined X and Y wind relative motion, so some change in tuning characteristics for combined axis movement is expected.

For most multi-rotors the bluff body drag is small compared to momentum drag for typical flight speeds so this change is not expected to make much difference to the tuning of exisiting setups.

A back to back comparison the legacy code using ekf replay on a tuned setup is recommended.

### CODE REVIEW

File: src/modules/ekf2/EKF/drag_fusion.cpp [hunk 1]
```diff
@@ -136,7 +136,7 @@ void Ekf::fuseDrag(const dragSample &drag_sample)
 				// The airspeed used for linearisation is calculated from the measured acceleration by solving the following quadratic
 				// mea_acc = 0.5 * rho * bcoef_inv * airspeed**2 + mcoef_corrrected * airspeed
 				const float airspeed = (bcoef  / rho) * (- mcoef_corrrected  + sqrtf(sq(mcoef_corrrected) + 2.0f * rho * bcoef_inv * mea_acc.length()));
-				Kacc = fmaxf(1e-1f, rho * bcoef_inv * airspeed + mcoef_corrrected);
+				Kacc = fmaxf(1e-1f, cosf(AoS) * (rho * bcoef_inv * airspeed + mcoef_corrrected));
```
Comment (bresch) [inline src/modules/ekf2/EKF/drag_fusion.cpp:139]: Maybe using
```
cos(AoS) = rel_wind_body(0) / rel_wind_body_norm
and
sin(AoS) = rel_wind_body(1) / rel_wind_body_norm
with 
rel_wind_body_norm = rel_wind_body.norm()
```
is a bit less expensive.

File: src/modules/ekf2/test/test_EKF_drag_fusion.cpp [hunk 2]
```diff
@@ -0,0 +1,238 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2019 ECL Development Team. All rights reserved.
```
Comment (bresch) [inline src/modules/ekf2/test/test_EKF_drag_fusion.cpp:3]:
```suggestion
 *   Copyright (c) 2022 PX4 Development Team. All rights reserved.
```

File: src/modules/ekf2/test/test_EKF_drag_fusion.cpp [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+	_ekf_wrapper.enableGpsFusion();
+	_sensor_simulator.startGps();
+
+	// Apply parameter changes required to do drag fusion wind estimation
+	parameters *_params = _ekf->getParamHandle();
+	_params->bcoef_x = 0.0f;
+	_params->bcoef_y = 0.0f;
+	_params->mcoef = 0.15f;
+	_params->fusion_mode = 33;
```
Comment (bresch) [inline src/modules/ekf2/test/test_EKF_drag_fusion.cpp:93]: please use the EkfWrapper class to set and get parameters; especially for the fusion bitmask:
1. to set a bit in the aid mask: https://github.com/PX4/PX4-Autopilot/blob/deb938fceaa79b927ae0f622d8b2b8c4ad10d391/src/modules/ekf2/test/sensor_simulator/ekf_wrapper.cpp#L83-L86
1. to set a value in a parameter: https://github.com/PX4/PX4-Autopilot/blob/deb938fceaa79b927ae0f622d8b2b8c4ad10d391/src/modules/ekf2/test/sensor_simulator/ekf_wrapper.cpp#L98-L101

File: src/modules/ekf2/test/test_EKF_drag_fusion.cpp [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+
+	// simulate a vehicle that is hovering and tilting into wind
+
+	_ekf->set_is_fixed_wing(false);
```
Comment (bresch) [inline src/modules/ekf2/test/test_EKF_drag_fusion.cpp:97]:
```suggestion
```

**Review by bresch — CHANGES_REQUESTED** [review]

**Comment by bresch** [comment]
Found this while testing the drag fusion unit tests: https://github.com/PX4/PX4-Autopilot/pull/20072

File: src/modules/ekf2/test/test_EKF_drag_fusion.cpp (hunk 3 above)
Comment (priseborough) [reply src/modules/ekf2/test/test_EKF_drag_fusion.cpp:93]: Done

File: src/modules/ekf2/EKF/drag_fusion.cpp (hunk 1 above)
Comment (priseborough) [reply src/modules/ekf2/EKF/drag_fusion.cpp]: Done

File: src/modules/ekf2/test/test_EKF_drag_fusion.cpp (hunk 4 above)
Comment (priseborough) [reply src/modules/ekf2/test/test_EKF_drag_fusion.cpp]: Done

**Comment by bresch** [comment]
replaced by https://github.com/PX4/PX4-Autopilot/pull/20848

**Pull request closed without merging**